File: nagasenacs/src/Nagasena/Proc/Events/EXIEventNS.cs
using System.Diagnostics;

using BinaryDataSource = Nagasena.Proc.Common.BinaryDataSource;
using EventType = Nagasena.Proc.Common.EventType;
using EventDescription = Nagasena.Proc.Common.EventDescription;
using EventDescription_Fields = Nagasena.Proc.Common.EventDescription_Fields;
using Characters = Nagasena.Schema.Characters;

namespace Nagasena.Proc.Events {

  public sealed class EXIEventNS : EventDescription {

    private string m_uri; // "" represents disassociation
    private string m_prefix; // "" represents the default (i.e. no prefix)
    private bool m_localElementNs;
    private EventType m_eventType;

    public EXIEventNS(string prefix, string uri, bool localElementNs, EventType eventType) {
      Debug.Assert(eventType.itemType == EventType.ITEM_NS);

      m_prefix = prefix;
      m_uri = uri;
      m_localElementNs = localElementNs;
      m_eventType = eventType;
    }

    public sbyte EventKind {
      get {
        return EventDescription_Fields.EVENT_NS;
      }
    }

    public string URI {
      get {
        return m_uri;
      }
    }

    public string Name {
      get {
        return null;
      }
    }

    public int NameId {
      get {
        return -1;
      }
    }

    public int URIId {
      get {
        return -1;
      }
    }

    public bool LocalElementNs {
      get {
        return m_localElementNs;
      }
    }

    public EventType getEventType() {
      return m_eventType;
    }

    public string Prefix {
      get {
        return m_prefix;
      }
    }

    public Characters Characters {
      get {
        return null;
      }
    }

    public BinaryDataSource BinaryDataSource {
      get {
        return null;
      }
    }

  }

}

File: nagasena/src/org/openexi/proc/events/EXIEventNS.java
package org.openexi.proc.events;

import org.openexi.proc.common.BinaryDataSource;
import org.openexi.proc.common.EventType;
import org.openexi.proc.common.EventDescription;
import org.openexi.schema.Characters;

public final class EXIEventNS implements EventDescription {

  private String m_uri; // "" represents disassociation 
  private String m_prefix; // "" represents the default (i.e. no prefix) 
  private boolean m_localElementNs;
  private EventType m_eventType;
  
  public EXIEventNS(String prefix, String uri, boolean localElementNs, EventType eventType) {
    assert eventType.itemType == EventType.ITEM_NS;

    m_prefix = prefix;
    m_uri = uri;
    m_localElementNs = localElementNs;
    m_eventType = eventType;
  }
  
  public byte getEventKind() {
    return EventDescription.EVENT_NS;
  }

  public String getURI() {
    return m_uri;
  }

  public String getName() {
    return null;
  }
  
  public int getNameId() {
    return -1;
  }

  public int getURIId() {
    return -1;
  }
  
  public boolean getLocalElementNs() {
    return m_localElementNs;
  }

  public EventType getEventType() {
    return m_eventType;
  }

  public String getPrefix() {
    return m_prefix;
  }

  public Characters getCharacters() {
    return null;
  }

  public BinaryDataSource getBinaryDataSource() {
    return null;
  }
  
}
